bin/release-bom

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1090,SC1091
+set -eo pipefail
+
+help() {
+  echo "Description:"
+  echo "    Generate and publish a BOM JSON file for the given version to the public release repository."
+  echo
+  echo "Usage:"
+  echo "    ${0##*/} [-x] <version>"
+}
+
+while getopts ':hx' option ; do
+  case "${option}" in
+    h) help ; exit ;;
+    x) set -x ;;
+    *) help ; exit 1 ;;
+  esac
+done
+shift $((OPTIND - 1))
+
+if [[ "${#@}" -eq 0 ]] ; then
+  echo "ERROR: missing version argument"
+  echo
+  help
+  exit 1
+fi
+
+version="$1"
+repo_url="https://updates.astronomer.io/astronomer-software/releases"
+json_file_name="astronomer-${version}.json"
+local_json_file_path="./astronomer_bom_generator/doc/${json_file_name}"
+max_age="300"
+
+# Step 1: Runs the Python extractor script to generate the BOM JSON file
+echo "Running the extractor script to generate ${json_file_name}..."
+pip install git+https://"${GITHUB_TOKEN}"@github.com/astronomer/astronomer-bom-generator.git@v0.1.0
```

Comment:
While this syntax does work, it's better to quote the whole string.

Also we should move the bom-generator version into a variable:
```suggestion
pip install "git+https://${GITHUB_TOKEN}@github.com/astronomer/astronomer-bom-generator.git@${BOM_GENERATOR_VERSION}"
```

We'll need to set `BOM_GENERATOR_VERSION=v0.1.0` higher up in the script. Doing it this way makes it easy to `git grep _VERSION=` to see what all our managed component versions are assigned to when it comes time to update versions.

Reply:
@Shubham-astro it looks like you merged the code suggestion without also adding the second part I described.